astropy/coordinates/polarization.py

```diff
@@ -0,0 +1,252 @@
+from collections import namedtuple
+from contextlib import contextmanager
+from typing import Dict
+
+import numpy as np
+
+from astropy.utils import unbroadcast
+from astropy.utils.data_info import MixinInfo
+from astropy.utils.shapes import ShapedLikeNDArray
+
+__all__ = ["StokesCoord", "custom_stokes_symbol_mapping", "StokesSymbol"]
+
+
+StokesSymbol = namedtuple("StokesSymbol", ["symbol", "description"], defaults=[""])
+
+# This is table 29 in the FITS 4.0 paper
+FITS_STOKES_VALUE_SYMBOL_MAP = {
```

Comment:
This feels like an Enum. Would it be better as such?

Reply:
Enums are extended by subclassing, which we can do dynamically with `type()`. But that's not a normal usage of enum...
Besides the contextmanager, is there another way to register new symbols?

Reply:
Well you can just change the global variable...

Reply:
I think I would rather keep this as-is for now. Maybe this could be made to work if we transitioned it to using `ScienceState` as was suggested earlier in the review, but given where we are in the 5.3 release cycle I think keeping this the same as the WCS frame registration mechanism is what I would prefer.

Reply:
That sounds fine to me. An Enum has similar key uniqueness properties to a dict, so if it's decided to use an Enum instead the transition should be fairly painless (at least for users) by transforming the dict into an Enum.